dockerplugin.py

```diff
@@ -51,109 +51,106 @@ def _d(d):
-            blkio_stats = {}
-            for value in values:
-                k = '{key}-{major}-{minor}'.format(key=key,
-                                                   major=value['major'],
-                                                   minor=value['minor'])
-                if k not in blkio_stats:
-                    blkio_stats[k] = []
-                blkio_stats[k].append(value['value'])
-
-            for type_instance, values in blkio_stats.items():
-                if len(values) == 5:
-                    cls.emit(container, dimensions, 'blkio', values,
-                             type_instance=type_instance, t=t)
-                elif len(values) == 1:
-                    # For some reason, some fields contains only one value and
-                    # the 'op' field is empty. Need to investigate this
-                    cls.emit(container, dimensions, 'blkio.single', values,
-                             type_instance=key, t=t)
-                else:
-                    collectd.warn(('Unexpected number of blkio stats for '
-                                   'container {container}!')
-                                  .format(container=_c(container)))
-
-
-class CpuStats(Stats):
-    @classmethod
-    def read(cls, container, dimensions, stats, t):
-        cpu_usage = stats['cpu_usage']
-        percpu = cpu_usage['percpu_usage']
-        for cpu, value in enumerate(percpu):
-            cls.emit(container, dimensions, 'cpu.percpu.usage', [value],
-                     type_instance='cpu%d' % (cpu,), t=t)
-
-        items = sorted(stats['throttling_data'].items())
-        cls.emit(container, dimensions, 'cpu.throttling_data',
-                 [x[1] for x in items], t=t)
-
-        values = [cpu_usage['total_usage'], cpu_usage['usage_in_kernelmode'],
-                  cpu_usage['usage_in_usermode'], stats['system_cpu_usage']]
-        cls.emit(container, dimensions, 'cpu.usage', values, t=t)
-
-
-class NetworkStats(Stats):
-    @classmethod
-    def read(cls, container, dimensions, stats, t):
-        items = stats.items()
-        items.sort()
-        cls.emit(container, dimensions, 'network.usage',
-                 [x[1] for x in items], t=t)
-
-
-class MemoryStats(Stats):
-    @classmethod
-    def read(cls, container, dimensions, stats, t):
-        values = [stats['limit'], stats['max_usage'], stats['usage']]
-        cls.emit(container, dimensions, 'memory.usage', values, t=t)
-
-        for key, value in stats['stats'].items():
-            cls.emit(container, dimensions, 'memory.stats', [value],
+def emit(container, dimensions, type, value, t=None,
+         type_instance=None):
+    """Emit a collected datapoint."""
+    val = collectd.Values()
+    val.plugin = 'docker'
+    val.plugin_instance = container['Name']
+
+    # Add additional extracted dimensions through plugin_instance.
+    if dimensions:
+        val.plugin_instance += '[{dims}]'.format(dims=_d(dimensions))
+
+    if type:
```

Comment:
a variable shouldn't be named type due to shadowing a built-in function (type()) anyway